sycl/test/context-with-multiple-devices/context-with-multiple-devices.cpp

```diff
@@ -0,0 +1,46 @@
+// REQUIRES: cpu, accelerator, aoc
+
+// UNSUPPORTED: cuda, level_zero
+
+// RUN: %clangxx -fsycl -fsycl-unnamed-lambda -fsycl-targets=%sycl_triple %s -o %t1.out
+// RUN: %CPU_RUN_PLACEHOLDER CL_CONFIG_CPU_EMULATE_DEVICES=2 %t1.out
+// RUN: %CPU_RUN_PLACEHOLDER CL_CONFIG_CPU_EMULATE_DEVICES=4 %t1.out
+// RUN: %clangxx -fsycl -fintelfpga -fsycl-unnamed-lambda %s -o %t2.out
+// RUN: %ACC_RUN_PLACEHOLDER CL_CONFIG_CPU_EMULATE_DEVICES=2 %t2.out
+
+#include <CL/sycl.hpp>
+
+void exceptionHandler(sycl::exception_list exceptions) {
+  for (std::exception_ptr const &e : exceptions) {
+    try {
+      std::rethrow_exception(e);
+    } catch (sycl::exception const &e) {
+      std::cout << "Caught asynchronous SYCL exception:\n"
+                << e.what() << std::endl;
+    }
+  }
+}
+
+int main() {
+  std::vector DeviceList = sycl::device::get_devices();
```

Comment:
I would think this is a syntax error (when <device> is not specified). Hmm..., how/why does this work?

Reply:
The compiler compiles a code with C++17 enabled by default now (it's applicable only to user code, not to DPC++ RT, it's still on C++14), so as this test is kinda user code, I thought it would be good to use CTAD feature from C++17 here.